### PR TITLE
CPR-65 implement nomis search

### PR DESCRIPTION
**Pull request by cairnsp**
- New PrisonerService implementation and tests
- Integration of Nomis search into CourtCaseProcessor flow
- JUnit and Integration test updates

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/client/PrisonerSearchClientIntTest.kt [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+//    wireMockExtension
+//    wireMockServer.start()
+//    configureFor("localhost", 8090)
+//  }
+//
+//  @AfterEach
+//  fun tearDown() {
+//    wireMockServer.stop()
+//  }
+
```
Comment (markreesmoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/client/PrisonerSearchClientIntTest.kt:31]: Looks like we don't need this commented out code

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/PrisonerService.kt [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+          ),
+        )
+      }
+    }
+  }
+
+  private fun matchesExistingPrisonerPartially(prisoners: List<Prisoner>, person: Person): Boolean {
+    return prisoners.isNotEmpty()
+      .and(prisoners.size == 1)
+      .and(
```
Comment (markreesmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/PrisonerService.kt:55]: If the above checks are required here, are they also required for matchesExistingPrisonerExactly?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/PrisonerService.kt [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        prisoners.any {
+          it.pncNumber.equals(person.otherIdentifiers?.pncNumber) &&
+            (
+              it.firstName.equals(person.givenName, true) ||
+                it.lastName.equals(person.familyName, true) ||
+                it.dateOfBirth == person.dateOfBirth
```
Comment (markreesmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/PrisonerService.kt:61]: I am no Kotlin expert - curious why we do not use the equals method here?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/client/PrisonerSearchClientIntTest.kt (hunk 1 above)
Comment (vb-moj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/client/PrisonerSearchClientIntTest.kt:31]: Agree.

**Comment by vb-moj** [comment]
One minor comment.

**Review by vb-moj — APPROVED** [review]
LGTM

**Pull request merged**